### PR TITLE
MDP-184: Universal Popup Component

### PR DESCRIPTION
**Pull request by OlehVynnyk**
https://velocity.globallogic.com/jira/browse/MDP-184

### CODE REVIEW

File: src/components/InfoMessage/InfoMessage.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React, { FC } from "react";
+
+interface IProps {
+  text?: string;
```
Comment (pggl) [inline src/components/InfoMessage/InfoMessage.tsx]: text props is required always so remove ?

File: src/components/Message/Message.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React, { FC } from "react";
+
+interface IProps {
+  text?: string;
```
Comment (pggl) [inline src/components/Message/Message.tsx]: text is required

File: src/components/PopupButton/PopupButton.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React, { FC } from "react";
+
+interface IProps {
+  text?: string;
```
Comment (pggl) [inline src/components/PopupButton/PopupButton.tsx]: text is required

File: src/components/PopupButton/PopupButton.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+const ModalButton: FC<IProps> = ({ text, backgroundColor = "black" }) => (
+  <div className="flex justify-center">
+    <button className={`bg-${backgroundColor} h-10 w-5/6 rounded-lg  `}>
+      <div className="text-white font-lato text-base">{text}</div>
+    </button>
+  </div>
+);
+
+export default ModalButton;
```
Comment (pggl) [inline src/components/PopupButton/PopupButton.tsx]: component name is not same as filename

File: src/components/PopupsMobile/PopupsMobile.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import React, { FC } from "react";
+
+const StyledWrapper: React.CSSProperties = {
```
Comment (pggl) [inline src/components/PopupsMobile/PopupsMobile.tsx]: use better name for variable

File: src/components/PopupsMobile/PopupsMobile.tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  width: "343px",
+  height: "420px"
+};
+
+const Wrapper: React.CSSProperties = {
```
Comment (pggl) [inline src/components/PopupsMobile/PopupsMobile.tsx]: use destruction for CSSProperties

File: src/components/Testing.tsx [hunk 7]
```diff
@@ -1,24 +1,7 @@
...
-      <ButtonExample text="Nákup kreditov" backgroundColor="black" />
-      <ButtonExample text="Nákup kreditov" backgroundColor="orangeRed" />
-      <ButtonExample text="Nákup kreditov" backgroundColor="shamrockGreen" />
-      <div className="p-4 bg-red border-red-dark border border-solid font-gilbert">
-        test tailwind
-      </div>
-    </Wrapper>
-  </div>
-);
+const Testing: FC<RouteComponentProps> = () => <PopupsMobile />;
```
Comment (pggl) [inline src/components/Testing.tsx]: after development remove component so component will be empty

File: tailwind.js [hunk 8]
```diff
@@ -119,9 +119,9 @@ module.exports = {
   */
 
   fonts: {
-    "gilbert": ["Gilbert", "serif"],
+    gilbert: ["Gilbert", "serif"],
```
Comment (pggl) [inline tailwind.js]: discard all changes in this file

File: src/components/PopupsMobile/PopupsMobile.tsx [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+  minHeight: "380px"
+};
+
+const Modal: FC = props => (
+  <div style={StyledWrapper} className="pt-4 pb-6 border-black border m-auto">
+    <div style={Wrapper} className="flex flex-col justify-between">
```
Comment (pggl) [inline src/components/PopupsMobile/PopupsMobile.tsx]: is this div necessary?

**Review by pggl — CHANGES_REQUESTED** [review]

File: src/components/Modal/ModalButton.tsx [hunk 10]
```diff
@@ -0,0 +1,16 @@
+import React, { FC } from "react";
+
+interface IProps {
+  text?: string;
```
Comment (pggl) [inline src/components/Modal/ModalButton.tsx]: is text optional if so where is default value

File: src/components/Modal/ModalStorybook/DisprovePopup.tsx [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+      <InfoMessage text="BABY CHAT - 15:30, 28.január 2019" />
+    </div>
+    <div>
+      <ModalButton text="Nákup kreditov" backgroundColor="red" />
+      <TextUnderButton />
+    </div>
+  </Modal>
+);
+
+export default DisprovePopup;
```
Comment (pggl) [inline src/components/Modal/ModalStorybook/DisprovePopup.tsx]: I think cancel reservation is better name

File: src/components/Modal/ModalStorybook/ErrorPopup.tsx [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+      <InfoMessage text="BABY CHAT - 15:30, 28.január 2019" />
+    </div>
+    <div>
+      <ModalButton text="Nákup kreditov" backgroundColor="red" />
+      <TextUnderButton />
+    </div>
+  </Modal>
+);
+
+export default ErrorModal;
```
Comment (pggl) [inline src/components/Modal/ModalStorybook/ErrorPopup.tsx]: is ErrorModal same as DisprovePopup?

File: src/components/Modal/ModalStorybook/KreditPopup.tsx [hunk 13]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import XButton from "../XButton";
+import Message from "../Message";
+import InfoMessage from "../InfoMessage";
+import ModalButton from "../ModalButton";
+import Modal from "../MainModal";
```
Comment (pggl) [inline src/components/Modal/ModalStorybook/KreditPopup.tsx]: Rename it, we use Popup, we don't want to have to two names Modal and Popup

File: src/components/Modal/ModalStorybook/RegistrationPopup.tsx [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+      <InfoMessage text="BABY CHAT - 15:30, 28.január 2019" />
+    </div>
+    <div>
+      <ModalButton text="Nákup kreditov" backgroundColor="green" />
+      <TextUnderButton />
+    </div>
+  </Modal>
+);
+
+export default ErrorModal;
```
Comment (pggl) [inline src/components/Modal/ModalStorybook/RegistrationPopup.tsx]: component name and filename is not same

File: src/components/Modal/TextUnderButton.tsx [hunk 15]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+
+const TextUnderbutton = () => (
```
Comment (pggl) [inline src/components/Modal/TextUnderButton.tsx]: we need props for text

File: src/styles/colors.ts [hunk 16]
```diff
@@ -0,0 +1,9 @@
+export const black = "#333333";
+export const white = "#ffffff";
+export const warmGrey = "#999696";
+export const maize = "#ffc450";
+export const aquaBlue = "#00b5e2";
+export const orangeRed = "#f93822";
+export const paleLilac = "#ebebec";
+export const shamrockGree = "#00c65e";
+export const pinkyPurple = "#db3eb1";
```
Comment (pggl) [inline src/styles/colors.ts]: why are those colors here? we have already defined colors in our app

**Review by pggl — CHANGES_REQUESTED** [review]

File: src/components/Popup/MainPopup.tsx [hunk 17]
```diff
@@ -0,0 +1,15 @@
+import React, { FC } from "react";
+
+const MainPopup: FC = props => (
```
Comment (garySiladi) [inline src/components/Popup/MainPopup.tsx]: could use destructuring props as `const MainPopup: FC = ({ children }) => (`

File: src/components/Popup/PopupButton.tsx [hunk 18]
```diff
@@ -0,0 +1,16 @@
+import React, { FC } from "react";
+
+interface IProps {
+  text: string;
+  backgroundColor?: "black" | "red" | "green";
+}
+
+const PopupButton: FC<IProps> = ({ text, backgroundColor = "black" }) => (
+  <div className="flex justify-center">
+    <button className={`bg-${backgroundColor} h-10 w-5/6 rounded-lg  `}>
```
Comment (garySiladi) [inline src/components/Popup/PopupButton.tsx]: delete space on the end of the classname

File: src/components/Popup/PopupStorybook/KreditPopup.tsx [hunk 19]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import XButton from "../XButton";
+import Message from "../Message";
+import InfoMessage from "../InfoMessage";
+import PopupButton from "../PopupButton";
+import MainPopup from "../MainPopup";
+
+const KreditPopup = () => (
```
Comment (garySiladi) [inline src/components/Popup/PopupStorybook/KreditPopup.tsx]: rename kredit to credit, as grammatically correct in english

File: src/components/Popup/XButton.tsx [hunk 20]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+
+const XButton = () => (
```
Comment (garySiladi) [inline src/components/Popup/XButton.tsx]: could rename this to `CloseButton`

**Review by garySiladi — CHANGES_REQUESTED** [review]

File: src/components/Popup/Storybook/CreditPopup.tsx [hunk 21]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import XButton from "../CloseButton";
+import Message from "../Message";
+import InfoMessage from "../InfoMessage";
+import Button from "../Button";
+import Index from "../Index";
+
+const KreditPopup = () => (
```
Comment (garySiladi) [inline src/components/Popup/Storybook/CreditPopup.tsx]: could you rename this to `CreditPopup`?

File: src/components/Popup/CloseButton.tsx [hunk 22]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+
+const CloseButton = () => (
+  <div className="flex justify-end ">
```
Comment (garySiladi) [inline src/components/Popup/CloseButton.tsx]: remove space in the end of className

File: src/components/Popup/CloseButton.tsx [hunk 23]
```diff
@@ -0,0 +1,11 @@
...
+    <button className=" bg-black font-bold w-10 h-10 rounded-lg mr-4 ">
```
Comment (garySiladi) [inline src/components/Popup/CloseButton.tsx]: remove space in the end of className

**Review by garySiladi — CHANGES_REQUESTED** [review]

**Review by garySiladi — DISMISSED** [review]
nice work

File: src/components/Popup/Index.tsx [hunk 24]
```diff
@@ -0,0 +1,13 @@
+import React, { FC } from "react";
+
+export const Index: FC = ({ children }) => (
```
Comment (pggl) [inline src/components/Popup/Index.tsx]: rename it to Popup, if there will be default it is ok to filename should be index.tsx but not with **capital letter**

**Review by pggl — CHANGES_REQUESTED** [review]

**Review by pggl — APPROVED** [review]
good work

**Pull request merged**